squash/dashboard/serializers.py

```diff
@@ -0,0 +1,72 @@
+from rest_framework import serializers
+from rest_framework.reverse import reverse
+from .models import Job, Metric, Measurement
+from django.db import transaction
+
+
+class MetricSerializer(serializers.ModelSerializer):
+
+    links = serializers.SerializerMethodField()
+
+    class Meta:
+        model = Metric
+        fields = ('metric', 'description', 'units', 'condition',
+                  'minimum', 'design', 'stretch', 'user', 'links',)
+
+    def get_links(self, obj):
+        request = self.context['request']
+        return {
+            'self': reverse('metric-detail', kwargs={'pk': obj.pk},
+                            request=request),
+         }
+
+
+class MeasurementSerializer(serializers.ModelSerializer):
+
+    class Meta:
+        model = Measurement
+        fields = ('metric', 'value',)
+
+    def get_links(self, obj):
+        request = self.context['request']
+        return {
+            'self': reverse('measurement-detail', kwargs={'pk': obj.pk},
+                            request=request),
+        }
+
+
+class JobSerializer(serializers.ModelSerializer):
+
+    links = serializers.SerializerMethodField()
+
+    measurements = MeasurementSerializer(many=True)
+
+    class Meta:
+        model = Job
+        fields = ('name', 'build', 'runtime',
+                  'url', 'status', 'measurements', 'links',)
+
+    """ Override the create method to create nested objects
+    from request data
+    """
+
+    def create(self, data):
+        measurements = data.pop('measurements')
```

Comment:
Measurements at this point are not correct.

```python
logging.debug(self.context['request']._request.body)
```
logging.debug(self.context['request']._request.body)

It looks like this:

```
2016-04-25 16:58:11,917 root-DEBUG [/Users/jmatt/dev/lsst/sqre/DM-5745/qa-dashboard/squash/dashboard/serializers.py 64] b'measurements=value&measurements=metric&url=https%3A%2F%2Fci.lsst.codes%2Fjob%2Fci_cfht%2F1%2F&status=0&name=ci_cfht&build=1&runtime=2016-04-24T19%3A26%3A12.561564Z'
```

Which does not include the actual value and metric value. I think you need to pass it as json. Which can be done like this (for testing purposes):

```python
resp = requests.post(api['job'], json=job, auth=(TEST_USER, TEST_PASSWD))
```